src/fudge/log.clj

```diff
@@ -20,6 +20,15 @@
   [c k & args]
   (apply (k c) c args))
 
+(defmacro calling-meta
+  "Used by other macros to get the namespace, file and line of the log call.
+   Builds a def with a generated symbol to expand and evaluate for the info."
```

Comment:
That's an implementation detail so possibly doesn't need to be in docstring.  Interesting though - is there any issue with having def's everywhere there's logging?  And should it be a `defonce` to avoid being reevaluated in loops?

Reply:
Done some testing with this. `defonce` *requires* a value, because it basically says "If the symbol has a value, leave it as-is. Otherwise set the symbol to arg #2". `(def sym#)` isn't actually setting the symbol to have a value, it's just defining the symbol as present. You can't do `(defonce sym#)`.